Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -531,16 +978,42 @@ public struct MongoConnectionString: Codable, LosslessStringConvertible {
     /// It is an error to specify both this option and `tlsDisableOCSPEndpointCheck`.
     public var tlsDisableCertificateRevocationCheck: Bool?
 
-    /// Indicates if OCSP responder endpoints should not be requested when an OCSP response is not stapled.
+    /// Specifies whether OCSP responder endpoints should not be requested when an OCSP response is not stapled.
     /// On macOS, this setting has no effect.
     /// By default this is set to false.
     public var tlsDisableOCSPEndpointCheck: Bool?
 
-    /// When specified, TLS constraints will be relaxed as much as possible. Currently, setting this option to `true`
+    /// Specifies whether TLS constraints will be relaxed as much as possible. Currently, setting this option to `true`
     /// is equivalent to setting `tlsAllowInvalidCertificates`, `tlsAllowInvalidHostnames`, and
     /// `tlsDisableCertificateRevocationCheck` to `true`.
     /// It is an error to specify both this option and any of the options enabled by it.
     public var tlsInsecure: Bool?
+
+    /// Specifies a WriteConcern to use for the client.
+    public var writeConcern: WriteConcern?
+}
+
+/// Helper extension to set a document field with a `MongoConnectionString.OptionName`.
+extension BSONDocument {
+    fileprivate subscript(name: MongoConnectionString.OptionName) -> BSON? {
+        get {
+            self[name.rawValue]
+        }
+        set {
+            self[name.rawValue] = newValue
+        }
+    }
+}
+
+extension Compressor {
+    fileprivate init(name: String, level: Int32?) throws {
```

Comment:
Including this here rather than in the implementation for `Compressor` because it relies upon validations made in the parser.